packages/shared/src/components/profile/SquadsList.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { ReactElement, useContext, useState } from 'react';
+import Link from 'next/link';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useRouter } from 'next/router';
+import request from 'graphql-request';
+import { SourceMember, SourceMemberRole } from '../../graphql/sources';
+import { largeNumberFormat } from '../../lib/numberFormat';
+import { Image } from '../image/Image';
+import SquadMemberBadge from '../squads/SquadMemberBadge';
+import { CardLink } from '../cards/Card';
+import PlusIcon from '../icons/Plus';
+import { ButtonSize, ButtonVariant } from '../buttons/common';
+import { Button } from '../buttons/ButtonV2';
+import {
+  useJoinSquad,
+  useToastNotification,
+  useViewSize,
+  ViewSize,
+} from '../../hooks';
+import { labels } from '../../lib';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { graphqlUrl } from '../../lib/config';
+import AuthContext from '../../contexts/AuthContext';
+import {
+  ProfileV2,
+  PUBLIC_SOURCE_MEMBERSHIPS_QUERY,
+} from '../../graphql/users';
+import { Connection } from '../../graphql/common';
+import { AuthTriggers } from '../../lib/auth';
+import { webappUrl } from '../../lib/constants';
+
+export interface SquadsListProps {
+  memberships?: Connection<SourceMember>;
+  userId: string;
+}
+
+function SquadItem({
+  membership,
+  loading,
+}: {
+  membership: SourceMember;
+  loading?: boolean;
+}): ReactElement {
+  const { user, showLogin } = useContext(AuthContext);
+  const router = useRouter();
+  const squad = membership.source;
+  const showJoin = !squad.currentMember?.role && !loading;
+  const { displayToast } = useToastNotification();
+
+  const { mutateAsync: joinSquad, isLoading } = useMutation(
+    useJoinSquad({ squad }),
+    {
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+      onSuccess: () => router.push(squad?.permalink),
+    },
+  );
+
+  const onJoin = async () => {
+    if (!user) {
+      showLogin({
+        trigger: AuthTriggers.JoinSquad,
+        options: {
+          onLoginSuccess: joinSquad,
+          onRegistrationSuccess: joinSquad,
+        },
+      });
+
+      return;
+    }
+
+    await joinSquad();
+  };
+
+  return (
+    <div
+      className="flex relative items-center p-2 first:ml-4 tablet:first:ml-0 w-40 tablet:w-auto bg-theme-float hover:bg-theme-hover rounded-2xl"
```

Comment:
`rounded-2xl` should be the numeric version not sure how much it comes to but something like `rounded-12`